### PR TITLE
remove tag from entry #1377

### PR DESCRIPTION
**Pull request by tcitworld**
#1377

### CODE REVIEW

File: src/Wallabag/CoreBundle/Controller/TagController.php [hunk 1]
```diff
@@ -56,6 +56,26 @@ public function addTagFormAction(Request $request, Entry $entry)
...
+     *
+     * @Route("/remove-tag/{entry}/{tag}", requirements={"entry" = "\d+", "tag" = "\d+"}, name="remove_tag")
+     *
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    public function removeTagFromEntry(Request $request, Entry $entry, Tag $tag)
+    {
+        $entry->removeTag($tag);
+        $em = $this->getDoctrine()->getManager();
+        $em->flush();
```
Comment (tcitworld) [inline src/Wallabag/CoreBundle/Controller/TagController.php]: How can I avoid having to flush maybe two times here ?

**Comment by tcitworld** [comment]
If @modos189 is interested to make my awful design look like something... ;-)

File: src/Wallabag/CoreBundle/Repository/TagRepository.php [hunk 2]
```diff
@@ -51,4 +51,20 @@ public function findAllTags($userId)
             ->getQuery()
             ->getResult();
     }
+
+    /**
+     * Used only in test case to get a tag for our entry.
+     *
+     * @return Tag
+     */
+    public function findOnebyEntryAndLabel($entry, $label)
```
Comment (j0k3r) [inline src/Wallabag/CoreBundle/Repository/TagRepository.php]: `By` instead of `by`. Also what do you think about `findOneByEntryAndTagLabel` instead?

Comment (tcitworld) [inline src/Wallabag/CoreBundle/Repository/TagRepository.php]: :+1:

File: src/Wallabag/CoreBundle/Tests/Controller/TagControllerTest.php [hunk 3]
```diff
@@ -65,4 +67,28 @@ public function testAddTagToEntry()
...
+            ->findOneByUsernameAndNotArchived('admin');
+
+        $tag = $client->getContainer()
+            ->get('doctrine.orm.entity_manager')
+            ->getRepository('WallabagCoreBundle:Tag')
+            ->findOnebyEntryAndLabel($entry, $this->tagName);
+
+        $client->request('GET', '/remove-tag/'.$entry->getId().'/'.$tag->getId());
+
+        $this->assertEquals(302, $client->getResponse()->getStatusCode());
```
Comment (j0k3r) [inline src/Wallabag/CoreBundle/Tests/Controller/TagControllerTest.php]: Could you retrieve the entry again just to validated that it's no more associated to the deleted tag?

File: src/Wallabag/CoreBundle/Controller/TagController.php (hunk 1 above)
Comment (j0k3r) [inline src/Wallabag/CoreBundle/Controller/TagController.php]: Hum I don't think you can.

**Comment by modos189** [comment]
https://github.com/wallabag/wallabag/pull/1664

**Comment by nicosomb** [comment]
:beer:

**Pull request merged**